packages/world/src/modules/keysintable/KeysInTableHook.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { IStoreHook } from "@latticexyz/store/src/IStore.sol";
+
+import { ResourceSelector } from "../../ResourceSelector.sol";
+
+import { MODULE_NAMESPACE, KEYS_LENGTH_NAMESPACE, USED_KEYS_NAMESPACE } from "./constants.sol";
+import { KeysInTable } from "./tables/KeysInTable.sol";
+import { KeysInTableLength } from "./tables/KeysInTableLength.sol";
+import { UsedKeysIndex, UsedKeysIndexData } from "./tables/UsedKeysIndex.sol";
+import { ArrayLib } from "../utils/ArrayLib.sol";
+import { getTargetTableSelector } from "../utils/getTargetTableSelector.sol";
+
+/**
+ * This is a very naive and inefficient implementation for now.
+ * We can optimize this by adding support for `setIndexOfField` in Store
+ * and then replicate logic from solecs's Set.sol.
+ * (See https://github.com/latticexyz/mud/issues/444)
+ *
+ * Note: if a table with composite keys is used, only the first key is indexed
+ */
+contract KeysInTableHook is IStoreHook {
+  using ArrayLib for bytes32[];
+  using ResourceSelector for bytes32;
+
+  function handleSet(bytes32 tableId, bytes32[] memory key) internal {
+    bytes32 keysInTableTableId = getTargetTableSelector(MODULE_NAMESPACE, tableId);
+    bytes32 keysInTableLengthTableId = getTargetTableSelector(KEYS_LENGTH_NAMESPACE, tableId);
+    bytes32 usedIndexTableId = getTargetTableSelector(USED_KEYS_NAMESPACE, tableId);
```

Comment:
same here, why not use a single UsedIndex table with `{ sourceTable: "bytes32", keyHash: "bytes32" }` primary key?